### DISCUSSION TITLE
Plant Identification Model with EZKL

### DISCUSSION THREAD

**Discussion by Oba-One** [zkonduit/ezkl#149] [Q&A] [answered]
Hi, 

I'm working on a project called [WEFA](https://wefa.app) where it incentivizes people to care for plants by earning digital rewards to play AR games. A key piece is the ML plant identification and health assessment to determine if the plant chosen is in the curated list and right zone along with health wise how much to reward a user. We're currently looking to use [plant.id](https://plant.id/)'s API, ideally it would be nice to run the models on chain though we're not sure how feasible that is and how to source the models that would be compatible. We have a list of 16 plants with 4 in each category (herbs, fruits, flowers, vegetables) so potentially the model can be split into 4 pieces or even 1 for each plant. I don't have a ton of experience with models mainly using APIs for ML functionality but I'm willing and wanting to learn more with a dedicated use case. 

Would love any thoughts or feedback thanks!

**Comment by alexander-camuto** [comment, marked as answer]
Hey ! Probably best to create a single classifier across all categories (so 4*16 classes if I've understood correctly). Once you've built the model in pytorch you can run `ezkl` on it to create a model that can submit proofs on-chain + a smart contract that can verify said proofs :)

**Reply by Oba-One** [reply]
Thanks for the help! You have an example I could follow?

**Reply by Oba-One** [reply]
Also any recommendations on where to source models, specifically in the plant domain?

**Reply by alexander-camuto** [reply]
If its your own data, you're going to have to build the model and train it yourself :) I'd check out [sklearn](https://scikit-learn.org/stable/) though to experiment
